project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -244,8 +244,9 @@ sealed interface ProjectUpdateResponseBody permits FundingInformation, ProjectCo
 
   }
 
-  sealed interface ExperimentUpdateRequestBody permits ConfoundingVariables, ExperimentDescription,
-      ExperimentalGroups, ExperimentalVariables {
+  sealed interface ExperimentUpdateRequestBody permits ConfoundingVariableAdditions,
+      ConfoundingVariableDeletions, ConfoundingVariableUpdates, ExperimentDescription,
+      ExperimentalGroups, ExperimentalVariableAdditions, ExperimentalVariableDeletions {
 
   }
```

Comment:
Since i'm not up to date i assume that i missed something so here's my question:

What's the advantage of having an interface that could be any of these possibilities? 
As i saw below, this leads to the burden of the service implementation to account for all cases individually. 
Could this be seperated into individual use cases to avoid an extending switch statement?

Reply:
It could be separated into different methods which would make it easier to specify what should be done with the input. Deciding based on what was sent works too.